src/sage/groups/libgap_mixin.py

```diff
@@ -963,3 +967,115 @@ def is_isomorphic(self, H):
             (False, False, False)
         """
         return self.gap().IsomorphismGroups(H.gap()) != libgap.fail
+
+
+def minimum_generating_set(G) -> list:
+    """
+    Return a list of the minimum generating set of ``G``.
+
+    INPUT:
+
+    - ``G`` -- a group
+
+    OUTPUT:
+
+    A list of GAP objects that generate the group.
+
+    .. SEEALSO::
+
+        :meth:`sage.categories.groups.Groups.ParentMethods.minimum_generating_set`
+
+    ALGORITHM:
+
+    We follow :doi:`10.1016/j.jalgebra.2023.11.012` (:arxiv:`2306.07633`).
+
+    TESTS:
+
+    Test that the resultant list is able to generate the original group::
+
+        sage: from sage.groups.libgap_mixin import minimum_generating_set
+        sage: p = libgap.eval("DirectProduct(AlternatingGroup(5),AlternatingGroup(5))")
+        sage: s = minimum_generating_set(p); s
+        [(1,2,3,4,5)(8,9,10), (3,4,5)(6,7,8)]
+        sage: set(p.AsList()) == set(libgap.GroupByGenerators(s).AsList())
+        True
+
+    Test that elements of resultant list are GAP objects::
+
+        sage: from sage.groups.libgap_mixin import minimum_generating_set
+        sage: G = PermutationGroup([(1,2,3), (2,3), (4,5)])
+        sage: s = minimum_generating_set(G); s
+        [(2,3), (1,3,2)(4,5)]
+        sage: s[0].parent()
+        C library interface to GAP
+    """
+    if not isinstance(G, GapElement):
+        try:
+            G = G.gap()
+        except (AttributeError, ValueError, TypeError):
+            raise NotImplementedError("only implemented for groups that can construct a gap group")
+
+    if not G.IsFinite().sage():
+        raise NotImplementedError("only implemented for finite groups")
+
+    def is_group_by_gens(group, gens):
+        return set(group.AsList()) == set(libgap.GroupByGenerators(gens).AsList())
+
+    group_elements = G.AsList()
+
+    if G.IsCyclic().sage():
+        for ele in group_elements:
+            if is_group_by_gens(G, [ele]):
+                return [ele]
+
+    if G.IsSimple().sage():
+        n = len(group_elements)
+        for i in range(n):
+            for j in range(i+1, n):
+                if is_group_by_gens(G, [group_elements[i], group_elements[j]]):
+                    return [group_elements[i], group_elements[j]]
+
+    N = G.MinimalNormalSubgroups()[0]
```

Comment:
This is rather expensive... To give a simple benchmark, consider the group `DirectProduct(AlternatingGroup(50),AlternatingGroup(50))`.

The GAP function `MinimalGeneratingSet` (which, in your terminology, computes a "minimum" generating set) takes 2.7 seconds on my laptop to yield a generating set of size 2 (since this group is non-abelian, that proves it is indeed minimal).

In contrast, computing the minimal normal subgroups hasn't finished running after a few minutes.

Of course you don't need all minimal normal subgroups, just one. I haven't read your paper in detail but a quick glance reveals that you actually work with a chief series. Computing that is much cheaper; `ChiefSeries(G);` terminates in 2.5 seconds for that example group. So you may wish to tune your algorithm a bit.

Reply:
I tried the algorithm with the below implementation of the minimal normal subgroup, which finds only one minimal normal subgroup, but this also takes almost the same time as before. 
```python
    # Find minimal normal subgroup N of G
    L = list(libgap.Elements(G))
    if len(L)==1:
        return None
    x = L[1]
    o = int(libgap.Order(x))
    p = 2
    q = None
    while p**2 <= o:
        if o%p == 0:
            q = o//p
            break
        p += 1
    if q is None:
        q = 1
        p = o
    x = x**q
    N = G.Subgroup([x])
    N = libgap.NormalClosure(G,N)
```



> So you may wish to tune your algorithm a bit.

This implementation is the most straightforward polytime implementation proposed by the paper. We can consider future optimisations, but can we go with this implementation for now because it works for almost every group? If I change the implementation, I am not sure what will be the time taken by the new implementation (the one using the chief series).

Reply:
@RuchitJagodara, you should follow @fingolfin's advice. You should also profile your code to see where time is being spent. Right now, GAP's implementation is apparently completely beating yours, so it should be the preferred one unless you have data suggesting otherwise.